rustls/src/server/tls13.rs

```diff
@@ -880,23 +880,37 @@ impl ExpectFinished {
         })
     }
 
-    fn emit_stateless_ticket(
+    fn emit_ticket(
         handshake: &mut HandshakeDetails,
         sess: &mut ServerSessionImpl,
         key_schedule: &KeyScheduleTraffic,
     ) {
         let nonce = rand::random_vec(32);
         let plain = get_server_session_value(handshake, key_schedule, sess, &nonce).get_encoding();
-        let ticket = match sess.config.ticketer.encrypt(&plain) {
-            Some(t) => t,
-            None => return,
+
+        let (ticket, lifetime, stateful) = if sess.config.ticketer.enabled() {
+            let ticket = match sess.config.ticketer.encrypt(&plain) {
+                Some(t) => t,
+                None => return,
+            };
+            (ticket, sess.config.ticketer.get_lifetime(), false)
+        } else {
```

Comment:
I wonder if we should keep supporting stateful tickets in TLS 1.3. Maybe TLS 1.3 should be stateless only? Would implementing the server side of #466 push us in the direction of stateless only? After all, we don't want to store 10x the state on the server to be able to give the client 10 sessions.

Reply:
That seems out of scope for this PR, which is just about deduplicating some code. File a separate issue?

Reply:
If we drop the stateful resumption then we'd just do that instead of reviewing this PR, right? That's why I brought it up now. It seems like we're likely to be refactoring this to just immediately(?) throw it away.

Reply:
I suppose so. On the other hand, while there has been little discussion about dropping stateful ticket support, it still seems worthwhile to me to reduce the amount of duplicate code, potentially making future refactoring easier.

Reply:
It does seem like dropping the stateful session support would not be clearly a good idea anyway so I agree we should just review this.